plugins/inputs/windows_event_log/wineventlog/utils_test.go

```diff
@@ -77,6 +77,43 @@ func TestFullBufferUsedWithHalfUsedSizeReturned(t *testing.T) {
 	assert.Equal(t, bufferUsed, len(str))
 }
 
+func TestInsertPlaceholderValues(t *testing.T) {
+	evtDataValues := []Data{
+		{"value_1"}, {"value_2"}, {"value_3"}, {"value_4"},
+	}
+	tests := []struct {
+		name     string
+		message  string
+		expected string
+	}{
+		{
+			"Placeholders %{number} should be replaced by insertion strings",
+			"Service %1 in region %3 stop at %2",
+			"Service value_1 in region value_3 stop at value_2",
+		},
+		{
+			"Index should start from 1 and less than or equal to the amount of values in event data",
+			"%0 %3 %5",
+			"%0 value_3 %5",
+		},
+		{
+			"Handle consecutive % characters",
+			"%1 %%3% %2",
+			"value_1 %value_3% value_2",
+		},
+		{
+			"Handle % character at the end of message",
+			"%3 %2%",
+			"value_3 value_2%",
+		},
```

Comment:
I don't think `insertPlaceholderValues` will work for message = `"Service %foo1 has stopped"`.

Reply:
That's true. I was thinking checking `%` would iterate through the strings first and iterate again when inserting values. This would make it a O(2n) time. But I guess although it would be O(n) without checking `%`, re-constructing the string for `rawMessage` without `%` also takes time/resouce

Reply:
true true...will add more test cases